### PR TITLE
Normalize label-space capitalization and fix mutli-hot label numerical stability

### PR DESCRIPTION
**Pull request by josauder**
- Normalize source/target label-mapping matching to be case-insensitive and de-duplicate target labels (registry.py + test).
- Rename label id 0 from "background" to "ignore", because we added an explicit background class
- Concept-bottleneck loss now runs in logit space (cross-entropy / BCE-with-logits) for numerical stability under AMP via a new ConceptBottleneckOutput carrying raw concept_logits.

Small changes:
- Support training/validating Coralscapes V2 on multiple HF splits (e.g. train+validation) from the data config.
- DPTHead infers its patch-token grid from sequence length, enabling arbitrary input resolutions.
- Update demo assets (concept/id label maps) top match fixed label/concept mapping

### CODE REVIEW

File: mermaidseg/dataset_reconciliation/registry.py [hunk 1]
```diff
@@ -192,6 +196,10 @@ def __init__(
                         "label_roll_up=True requires benthic_hierarchy or fetch_remote=True"
                     )
                 benthic_hierarchy = initialize_benthic_hierarchy()
+            benthic_hierarchy = {
```
Comment (ViktorDomazetoski) [inline mermaidseg/dataset_reconciliation/registry.py]: should we just move this to the initialize_benthic_hierarchy() function itself?

File: mermaidseg/dataset_reconciliation/registry.py [hunk 2]
```diff
@@ -86,8 +87,9 @@ class SourceLabelRegistry:
         source_to_target_name_maps: Optional mapping
             ``{SOURCE_NAME: {source_label_name: target_label_name}}``. Defaults
             are looked up per ``SOURCE_NAME``: ``mermaid`` is identity,
-            ``coralscapes`` uses the static dict, ``coralnet`` and
-            ``catlin_seaview`` are fetched from the MERMAID API.
+            ``coralscapes``, ``coralscapes_v2`` ``catlin_seaview``, ``moorea_labeled_corals``, ``ucsd_mosaics``, ``benthos_yuval`` use the static dict,
+            ``coralnet`` and
+             are fetched from the MERMAID API.
```
Comment (ViktorDomazetoski) [inline mermaidseg/dataset_reconciliation/registry.py]: update to `coralnet`` is fetched from the MERMAID API. (typo nitpicking sorry)

File: mermaidseg/model/concept_metrics.py
Comment (ViktorDomazetoski) [inline mermaidseg/model/concept_metrics.py]: Might be wrong here, but can we simplify this by using the class BCEWithLogitsLoss in the case where from_logits=True (instead of the functional), and BCELoss in the case where from_logits=False, to simplify things? Then concept_loss can always be defined and per_element_loss can always be calculated in an unified way

File: mermaidseg/model/models.py [hunk 3]
```diff
@@ -283,7 +302,12 @@ def forward(self, x: torch.Tensor, labels=None, **kwargs: Any) -> SemanticSegmen
         #     loss_fct = torch.nn.CrossEntropyLoss(ignore_index=0)
         #     loss = loss_fct(logits.squeeze(), labels.squeeze())
 
-        return SemanticSegmenterOutput(loss=loss, logits=logits, hidden_states=concept_outputs)
+        return ConceptBottleneckOutput(
+            loss=loss,
+            logits=logits,
+            hidden_states=concept_outputs,
```
Comment (ViktorDomazetoski) [inline mermaidseg/model/models.py]: I kept the concept_outputs within the hidden_states parameter as sort of a hack (though it might not really be the proper meaning/usage for that parameter), so now that we have a proper ConceptOutput class (which I've also been meaning to implement so kudos haha), I'd keep the concept_outputs in a more appropriately named parameter unless we have some advantage from keeping it here (e.g. from the SemanticSegmenterOutput class ancestry). 

The second question I have is whether we want to keep both of them esp since they are pretty big tensors and like this take a lot of GPU space (esp if we are only using one?)

**Comment by ViktorDomazetoski** [comment]
Also update tests so that it passes:

From what I can see you need to update "background" to "ignore" in test_dataset_stats.py line 152-153:

```
assert counts["target_name"].tolist() == ["background", "Acropora", "Porites", "Other"]
assert counts["class_kind"].tolist() == [
    "background",
    "target",
    "target",
    "unclassified",
]
```

**Comment by ViktorDomazetoski** [comment]
Everything else looks good to me, so happy to merge once above is finished!

File: mermaidseg/model/meta.py [hunk 4]
```diff
@@ -319,9 +319,10 @@ def batch_predict_loss(
                 "target_concepts must be provided in 'concept-bottleneck' mode"
             )
             concept_outputs = segmentation_outputs.hidden_states.float()
+            concept_logits = segmentation_outputs.concept_logits.float()
```
Comment (ViktorDomazetoski) [inline mermaidseg/model/meta.py:376]: Can we also apply these same changes to the batch_predict function

File: mermaidseg/dataset_reconciliation/registry.py (hunk 1 above)
Comment (josauder) [reply mermaidseg/dataset_reconciliation/registry.py]: ✅

File: mermaidseg/dataset_reconciliation/registry.py (hunk 2 above)
Comment (josauder) [reply mermaidseg/dataset_reconciliation/registry.py]: ✅

File: mermaidseg/model/concept_metrics.py
Comment (josauder) [reply mermaidseg/model/concept_metrics.py]: Now always goes through a module

File: mermaidseg/model/models.py (hunk 3 above)
Comment (josauder) [reply mermaidseg/model/models.py]: Addressed: ConceptBottleneckOutput now has an explicit concept_outputs field; activations no longer go in hidden_states. Both concept_outputs (activations, for class head / inference) and concept_logits (for loss) are kept, with a docstring explaining why.

Re: GPU memory concern — we kept both tensors intentionally: logits are needed for stable logit-space loss; activations feed the classifier and demo inference. Happy to revisit if we want a memory-optimized path later.

File: mermaidseg/model/meta.py (hunk 4 above)
Comment (josauder) [reply mermaidseg/model/meta.py:376]: ✅

**Comment by ViktorDomazetoski** [comment]
LGTM

**Comment by yeelauren** [comment]
Yay, merging 🪂

**Review by yeelauren — APPROVED** [review]
